app/src/main/java/ch/epfl/favo/common/DatabaseWrapper.java

```diff
@@ -45,17 +53,50 @@ public static String generateRandomId() {
     return sb.toString();
   }
 
-  public static void addDocument(String key, Map document, String collection)
+  public static void addDocument(String key, Map document, String collection) {
+    new TaskToFutureAdapter<>(getCollectionReference(collection).document(key).set(document));
+  }
+
+  public static void removeDocument(String key, String collection) {
+    Task<Void> deleteTask = getCollectionReference(collection).document(key).delete();
+    new TaskToFutureAdapter<>(deleteTask);
```

Comment:
If you do not plan to use the future, then there is no need to adapt it. You can simply do the call above without retrieving the task back. The semantic is then "the document will be removed at some point in the future, but you do not know them and are not notified if it it fails", which is totally fine depending on the usage.

i.e. just call `getCollectionReference(collection).document(key).set(document)`